### ISSUE TITLE
🐞 [BUG] - /config/{id} and web UI report proxy online while /metrics reports status=0

### ISSUE THREAD

**Issue by xatang** [kutovoys/xray-checker#172]
**Describe the bug**

The `/config/{stable_id}` endpoint and the web UI report a proxy as online (HTTP 200 / green status) while the Prometheus `/metrics` endpoint correctly reports the same proxy as failed (`xray_proxy_status = 0`, `xray_proxy_latency_ms = 0`) at the exact same moment. The status returned by `/config/{id}` and shown in the dashboard does not match the actual check result exposed in `/metrics`. This makes `/config/{id}` unreliable for uptime monitoring (e.g. Uptime Kuma), since a down proxy is reported as up.

**To Reproduce**

1. Run Xray Checker with `PROXY_CHECK_METHOD=status` and multiple proxies sharing the same `name` from the same subscription (several `stable_id` values, same display name).
2. Wait until one of those proxies fails its check (so its metric becomes `xray_proxy_status ... 0`).
3. Query the metrics endpoint and the config endpoint for that exact `stable_id` at the same time:
   - `curl -s http://<host>:2112/metrics | findstr <stable_id>`
   - `curl -i -s http://<host>:2112/config/<stable_id>`
4. Observe that `/metrics` shows status `0` while `/config/{id}` returns `HTTP/1.1 200 OK`, and the web UI shows the proxy as online.

**Expected behavior**

`/config/{stable_id}` should return HTTP 503 (and the web UI should show the proxy as offline) whenever the corresponding `xray_proxy_status` metric for that `stable_id` is `0`. The status reported by the UI, the `/config/{id}` endpoint, and `/metrics` should always be consistent and reflect the latest check result.

**Screenshots**

Metrics output and config endpoint response captured at the same time:

```
> curl -s http://example-host:2112/metrics | findstr aaaa111122223333
xray_proxy_latency_ms{address="node.example.com:443",group_name="",name="Example-Vless-443",protocol="vless",stable_id="aaaa111122223333",sub_name="a.example.com"} 0
xray_proxy_status{address="node.example.com:443",group_name="",name="Example-Vless-443",protocol="vless",stable_id="aaaa111122223333",sub_name="a.example.com"} 0

> curl -i -s http://example-host:2112/config/aaaa111122223333
HTTP/1.1 200 OK
Date: Tue, 30 Jun 2026 09:19:36 GMT
Content-Length: 2
Content-Type: text/plain; charset=utf-8
OK
```

The dashboard also shows all three `Example-Vless-443` entries with a green (online) indicator at the same time, even though `stable_id="aaaa111122223333"` has `xray_proxy_status = 0` in metrics.

**Additional context**

- Version: v1.3.0
- Deployment: Docker, running alongside Home Assistant
- `PROXY_CHECK_METHOD=status`, interval 60s, timeout 30s
- Multiple proxies share the same `name` (`Example-Vless-443`) but have different `stable_id` values and different `sub_name` (`a.example.com`, `b.example.com`, `c.example.com`). The failing one is `sub_name="a.example.com"`.
- `/metrics` appears to be the only endpoint reflecting the true check result; `/config/{id}` and the UI appear to use a different/stale source.

**Comment by xatang** [comment]
## Root cause

Check results are stored in the checker's `results` map keyed by the **full Prometheus label set** — `protocol + address + name + sub_name + stable_id + group_name` (see `proxyMetricKey` in `checker/checker.go`). The `/metrics` endpoint renders each series by looking results up with this exact key, so it is always correct.

However, `/config/{stable_id}`, the web UI, and the JSON API resolve a proxy's status **by `name` only**, via `GetProxyResult(name)` / `GetProxyStatus(name)`. That function loops over the proxy list and returns the result of the **first** proxy whose `name` matches:

```go
for _, proxy := range pc.proxies {
    if proxy.Name == name {
        metricKey = proxyMetricKey(proxy)
        found = true
        break   // <-- first match wins
    }
}
```

When several proxies share the same `name` (e.g. three `Example-Vless-443` entries from different subscriptions, same display name, different `stable_id` / `sub_name`), this lookup is **ambiguous**: it returns whichever same-named proxy happens to come first in the list, ignoring `stable_id` entirely.

So for the failing proxy `stable_id="aaaa111122223333"`:
- `/metrics` reads its result by the unique full key → correctly reports `xray_proxy_status = 0`.
- `/config/aaaa111122223333` resolves the proxy by `stable_id`, then calls `GetProxyStatus(proxy.Name)` — which discards the `stable_id` and returns the status of the *first* `Example-Vless-443` (a healthy sibling) → `HTTP 200 OK`.

The web dashboard and `/api/v1/...` handlers have the same defect (they call `GetProxyResult(proxy.Name)`), which is why all three same-named entries show a green indicator simultaneously.

## Why it is not protocol-specific

The bug lives entirely in the name-keyed lookup, with no dependency on the protocol. VLESS only stood out in this report because those proxies shared a display name; Hysteria is affected identically whenever two or more Hysteria proxies share the same `name`. Any group of proxies with a duplicate `name` will mismatch between `/metrics` and `/config/{id}` + UI.

## Affected code paths

- `/config/{stable_id}` → `ConfigStatusHandler` → `GetProxyStatus(found.Name)` (`web/handlers.go`)
- Web UI status → `RegisterConfigEndpoints` → `GetProxyStatus(proxy.Name)` (`web/handlers.go`)
- JSON API → `GetProxyResult(proxy.Name)` / `GetProxyStatus(proxy.Name)` (`web/api.go`: public proxies, proxies list, single proxy, status summary)

All of these should resolve the result by the unique key (`stable_id` / full label set) instead of by `name`, the same way `/metrics` does.

## Reproduction (minimal)

1. Two+ proxies with identical `name` but different `stable_id`.
2. Make the one that is **not first** in the list fail (or make the first one healthy and a later one fail).
3. `/metrics` shows `xray_proxy_status = 0` for the failing `stable_id`, while `/config/<that_stable_id>` returns `200 OK` and the UI shows it green.

**Comment by kutovoys** [comment]
@xatang Thanks for the report!

Reproduced it: /metrics keys results by stable_id, but /config/{id}, the dashboard and the API were looking them up by name — so with duplicate names they grabbed the first same-named proxy's status.

Fix is ready (they all resolve by stable_id now). Ships in the next release.
